In [1]:
import pandas as pd
import numpy as np

import pymysql
from sqlalchemy import create_engine

from sklearn.metrics import confusion_matrix, accuracy_score, precision_score, recall_score, ConfusionMatrixDisplay
from sklearn.metrics import classification_report, f1_score, cohen_kappa_score

import getpass
password = getpass.getpass()

········


In [2]:
connection_string = 'mysql+pymysql://root:'+password+'@localhost/sakila'
engine = create_engine(connection_string)

How many distinct (different) actors' last names are there?

In [3]:
data = pd.read_sql_query('Select distinct(last_name) as "Unique Last Names" From sakila.actor', engine)
data

Unique Last Names
0              AKROYD
1               ALLEN
2             ASTAIRE
3              BACALL
4              BAILEY
..                ...
116           WINSLET
117       WITHERSPOON
118              WOOD
119              WRAY
120         ZELLWEGER

[121 rows x 1 columns]

In [4]:
data.describe()

Unique Last Names
count                121
unique               121
top               AKROYD
freq                   1

121 unique names

Add an additional column day_type with values 'weekend' and 'workday' depending on the rental day of the week.

In [5]:
query ='''SELECT *, dayofweek(rental_date) as Weekday,dayname(rental_date) as Dayname, month(rental_date) as Month,
CASE WHEN DAYOFWEEK(r.rental_date) IN (1,7) THEN 'Weekend'
ELSE 'Workday'
END DayType
FROM rental as r'''
data = pd.read_sql_query(query, engine)
data.head(10)

rental_id         rental_date  inventory_id  customer_id  \
0          1 2005-05-24 22:53:30           367          130   
1          2 2005-05-24 22:54:33          1525          459   
2          3 2005-05-24 23:03:39          1711          408   
3          4 2005-05-24 23:04:41          2452          333   
4          5 2005-05-24 23:05:21          2079          222   
5          6 2005-05-24 23:08:07          2792          549   
6          7 2005-05-24 23:11:53          3995          269   
7          8 2005-05-24 23:31:46          2346          239   
8          9 2005-05-25 00:00:40          2580          126   
9         10 2005-05-25 00:02:21          1824          399   

          return_date  staff_id         last_update  Weekday    Dayname  \
0 2005-05-26 22:04:30         1 2006-02-15 21:30:53        3    Tuesday   
1 2005-05-28 19:40:33         1 2006-02-15 21:30:53        3    Tuesday   
2 2005-06-01 22:12:39         1 2006-02-15 21:30:53        3    Tuesday   
3 2005-06-03 01:43:41         2 2006-02-15 21:30:53        3    Tuesday   
4 2005-06-02 04:33:21         1 2006-02-15 21:30:53        3    Tuesday   
5 2005-05-27 01:32:07         1 2006-02-15 21:30:53        3    Tuesday   
6 2005-05-29 20:34:53         2 2006-02-15 21:30:53        3    Tuesday   
7 2005-05-27 23:33:46         2 2006-02-15 21:30:53        3    Tuesday   
8 2005-05-28 00:22:40         1 2006-02-15 21:30:53        4  Wednesday   
9 2005-05-31 22:44:21         2 2006-02-15 21:30:53        4  Wednesday   

   Month  DayType  
0      5  Workday  
1      5  Workday  
2      5  Workday  
3      5  Workday  
4      5  Workday  
5      5  Workday  
6      5  Workday  
7      5  Workday  
8      5  Workday  
9      5  Workday

Get all films with ARMAGEDDON in the title.

In [6]:
query ='''SELECT * FROM film WHERE title LIKE "%%ARMAGEDDON%%" '''
data = pd.read_sql_query(query, engine)
data.head(10)

film_id                  title  \
0       39        ARMAGEDDON LOST   
1      507    LADYBUGS ARMAGEDDON   
2      571       METAL ARMAGEDDON   
3      598    MOSQUITO ARMAGEDDON   
4      838  STAGECOACH ARMAGEDDON   
5      844      STEERS ARMAGEDDON   

                                         description  release_year  \
0  A Fast-Paced Tale of a Boat And a Teacher who ...          2006   
1  A Fateful Reflection of a Dog And a Mad Scient...          2006   
2  A Thrilling Display of a Lumberjack And a Croc...          2006   
3  A Thoughtful Character Study of a Waitress And...          2006   
4  A Touching Display of a Pioneer And a Butler w...          2006   
5  A Stunning Character Study of a Car And a Girl...          2006   

   language_id original_language_id  rental_duration  rental_rate  length  \
0            1                 None                5         0.99      99   
1            1                 None                4         0.99     113   
2            1                 None                6         2.99     161   
3            1                 None                6         0.99      57   
4            1                 None                5         4.99     112   
5            1                 None                6         4.99     140   

   replacement_cost rating                                   special_features  \
0             10.99      G                                           Trailers   
1             13.99  NC-17                                     Deleted Scenes   
2             26.99  PG-13               Trailers,Commentaries,Deleted Scenes   
3             22.99      G                                           Trailers   
4             25.99      R                            Trailers,Deleted Scenes   
5             16.99     PG  Trailers,Commentaries,Deleted Scenes,Behind th...   

          last_update  
0 2006-02-15 05:03:42  
1 2006-02-15 05:03:42  
2 2006-02-15 05:03:42  
3 2006-02-15 05:03:42  
4 2006-02-15 05:03:42  
5 2006-02-15 05:03:42

Get 10 the longest films.

In [7]:
query ='''SELECT title, length from film
Order by length Desc LIMIT 10'''
data = pd.read_sql_query(query, engine)
data.head(10)

title  length
0      DARN FORRESTER     185
1        POND SEATTLE     185
2       CHICAGO NORTH     185
3       MUSCLE BRIGHT     185
4        WORST BANGER     185
5         GANGS PRIDE     185
6  SOLDIERS EVOLUTION     185
7           HOME PITY     185
8   SWEET BROTHERHOOD     185
9      CONTROL ANTHEM     185

How many films include Behind the Scenes content?

In [11]:
query ='''SELECT COUNT(film_id) AS "Number of films with behind scenes"
FROM film
WHERE special_features LIKE "%%Behind the Scenes%%"'''
data = pd.read_sql_query(query, engine)
data.head()

Number of films with behind scenes
0                                 538

Which kind of movies (rating) have a mean duration of more than two hours?

In [12]:
query ='''SELECT rating, round(AVG(length),2) AS 'Average length of films per rating' FROM film
GROUP BY rating
HAVING round(AVG(length),2)>120'''
data = pd.read_sql_query(query, engine)
data.head()

rating  Average length of films per rating
0  PG-13                              120.44

Rank films by length (filter out the rows that have nulls or 0s in length column). In your output, only select the columns title, length, and the rank.

In [13]:
query ='''SELECT title, length FROM film
WHERE length IS NOT NULL OR length<> 0
ORDER BY length DESC'''
data = pd.read_sql_query(query, engine)
data

title  length
0          CHICAGO NORTH     185
1         CONTROL ANTHEM     185
2         DARN FORRESTER     185
3            GANGS PRIDE     185
4              HOME PITY     185
..                   ...     ...
995         ALIEN CENTER      46
996            IRON MOON      46
997        KWAI HOMEWARD      46
998     LABYRINTH LEAGUE      46
999  RIDGEMONT SUBMARINE      46

[1000 rows x 2 columns]